### PR TITLE
Bugfixes for RTL Comments

### LINKED ISSUE #1562: Workspace Scroll not working for ScratchBlockComments in RTL
### Expected Behavior

Workspace should scroll with things on the workspace. E.g. if something (a block or a comment) is dragged outside of the bounds of the workspace, the workspace should expand such that that element can be centered in the current view.

### Actual Behavior

The above does not work for scratch block comments in RTL.

### LINKED ISSUE #1700: Block comments in RTL don't have expected position
### Expected Behavior

_Please describe what should happen_
Comment is attached just to the right of the block in LTR:
screen shot 2018-08-24 at 11 55 43 am

In RTL , it should appear just to the left.

### Actual Behavior
screen shot 2018-08-24 at 11 56 34 am

### Steps to Reproduce

load https://scratch.mit.edu/projects/230672117/

### PR DESCRIPTION
**Pull request by kchadha**
### Resolves

Resolves #1700
Resolves #1562
Resolves issue corresponding to the scratch-blocks version of google/blockly#1896

### Proposed Changes

A number of issues corresponding to the use of comments in RTL have been resolved.
This includes: 
- Auto positioning of minimized block comments on a workspace loaded in RTL.
- Correctly positioning comments when switching back and forth between LTR and RTL workspaces:
    - The way scratch-gui does this, this required updates to block move events and comment move events so that they report the "language agnostic" position of the comment. This is a number that should not change regardless of LTR/RTL. This is the number reported in the XML for example.
- `getRelativeToSurfaceXY` now reports the position of the top right corner of the ScratchBlockComment in RTL. This is the same behavior as blocks.

### Reason for Changes

All of the issues above.

### Test Coverage

Manually tested in vertical playground and scratch-gui with this project: 230672117

Things to test (in scratch-gui):
- loading a project with comments in hebrew (e.g. via `?locale=he`)
- loading a project with comments in English and then switching to hebrew and back and forth a few times
- Moving comments in a workspace in either (LTR or RTL) and then switching to the other direction workspace
- Moving comments in a workspace in either direction and then undoing the move

### Known Issues
- In RTL, comments are have a really small offset from their position in LTR. This issue needs further investigation because it does not appear in the vertical playground.

### CODE REVIEW

File: core/block_events.js [hunk 1]
```diff
@@ -56,6 +56,7 @@ Blockly.Events.BlockBase = function(block) {
    */
   this.blockId = block.id;
   this.workspaceId = block.workspace.id;
+  this.workspace = block.workspace;
```
Comment (rachel-fenichel) [inline core/block_events.js]: Why did you add this?  Both places where you want the workspace already have a local variable representing the workspace, and at any rate only the workspace ID is serialized.

**Review by rachel-fenichel — APPROVED** [review]
LGTM.

**Comment by rachel-fenichel** [comment]
Please make this as a PR upstream as well.

**Review by chrisgarrity — APPROVED** [review]
Looks good - thanks for all the clarifying comments - it's great to know when it should be language aware, and when it shouldn't.

**Comment by kchadha** [comment]
Just updated the tests (which were failing because of the changes to events) and removed the unnecessary assignment of `this.workspace` as @rachel-fenichel noted in the review.

**Review by chrisgarrity — APPROVED** [review]
Tests look fine!

**Pull request merged**